tests/Doctrine/Tests/ORM/Mapping/Fixtures/AttributeEntityWithNestedJoinColumns.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Mapping\Fixtures;
+
+use Doctrine\ORM\Mapping as ORM;
+
+#[ORM\Entity]
+class AttributeEntityWithNestedJoinColumns
+{
+    /** @var int */
+    #[ORM\Id]
+    #[ORM\Column(type: 'integer')]
+    #[ORM\GeneratedValue]
+    public $id;
+
+    /** @var AttributeEntityWithNestedJoinColumns[] */
```

Comment:
This is not supposed to be an array, is it?

```suggestion
    /** @var Collection<AttributeEntityWithNestedJoinColumns> */
```

Reply:
Copy pasted from https://github.com/doctrine/orm/blob/2.15.x/tests/Doctrine/Tests/ORM/Mapping/AttributeDriverTest.php#L139. Does it make sense to change that there too?

Reply:
It would I suppose, but in a separate commit.